docs/source/pyopenms_in_r.rst

```diff
@@ -0,0 +1,243 @@
+
+    (possibly screenshot)
+
+    Cython signature: void load(String filename, libcpp_vector[ProteinIdentification] & protein_ids, libcpp_vector[PeptideIdentification] & peptide_ids)
+
+In this case the idXML$load() function requires
+
+       - a filename as string
+       - an empty vector for pyopenms.ProteinIdentification objects
+       - an empty vector for pyopenms.PeptideIdentification objects
+
+Creating an empty R list() unfortunately is not equal to the empty python list []
+
+Therefore in this case we need to use the reticulate::r_to_py() function:
+
+.. code-block:: R
+
+    f="/OpenMS/OpenMS/share/OpenMS/examples/BSA/BSA1_OMSSA.idXML"
+    pepids=r_to_py(list())
+    protids=r_to_py(list())
+
+    idXML$load(f, pepids, protids)
+
+In order to get more information about the wrapped functions, we can also 
+consult the `pyOpenMS manual <http://proteomics.ethz.ch/pyOpenMS_Manual.pdf>`_ 
+which references to all wrapped functions.
+
+An example use case
+*******************
+
+Reading an mzML File
+^^^^^^^^^^^^^^^^^^^^
+
+pyOpenMS supports a variety of different files through the implementations in
+OpenMS. In order to read mass spectrometric data, we can download the `mzML
+example file <http://proteowizard.sourceforge.net/example_data/tiny.pwiz.1.1.mzML>`_
+
+.. code-block:: R
+
+    from urllib.request import urlretrieve
+    # from urllib import urlretrieve  # use this code for Python 2.x
+    library(reticulate)
+    ropenms=import("pyopenms", convert = FALSE)
+    mzML=ropenms$mzMLFile()
+    urlretrieve ("http://proteowizard.sourceforge.net/example_data/tiny.pwiz.1.1.mzML", "tiny.pwiz.1.1.mzML")
+    exp = ropenms$MSExperiment()
+    mzML$load("tiny.pwiz.1.1.mzML", exp)
+
+which will load the content of the "tiny.pwiz.1.1.mzML" file into the ``exp``
+variable of type ``MSExperiment``.
+We can now inspect the properties of this object:
+
+.. code-block:: R
+
+    py_help(exp)
+    Help on MSExperiment object:
+
+    class MSExperiment(__builtin__.object)
+     |  Methods defined here:
+     ...
+     |  getNrChromatograms(...)
+     |      Cython signature: size_t getNrChromatograms()
+     |
+     |  getNrSpectra(...)
+     |      Cython signature: size_t getNrSpectra()
+     |
+     ...
+
+
+which indicates that the variable ``exp`` has (among others) the functions
+``getNrSpectra`` and ``getNrChromatograms``. We can now try these functions:
+
+.. code-block:: R
+
+    exp$getNrSpectra()
+    4
+    exp$getNrChromatograms()
+    2
+
+and indeed we see that we get information about the underlying MS data. We can
+iterate through the spectra as follows:
+
+Visualize spectra
+^^^^^^^^^^^^^^^^^
+
+You can easily visualise ms1 level precursor maps:
+
+.. code-block:: R
+
+    spectra = py_to_r(msexp$getSpectra())
+
+    ms1=sapply(spectra, function(x) x$getMSLevel()==1)
+    peaks=sapply(spectra[ms1], function(x) cbind(do.call("cbind", x$get_peaks()),x$getRT()))
+    peaks=do.call("rbind", peaks)
```

Comment:
this looks clever, but is maybe not easy to understand for a first-time R user. Maybe make a "traditional" loop and then explain in a second codeblock how it can be done with sapply?